ipfix/decoder.go

```diff
@@ -96,75 +98,115 @@ func NewDecoder(raddr net.IP, b []byte) *Decoder {
 
 // Decode decodes the IPFIX raw data
 func (d *Decoder) Decode(mem MemCache) (*Message, error) {
-	var (
-		msg = new(Message)
-		err error
-	)
+	var msg = new(Message)
 
 	// IPFIX Message Header decoding
-	if err = msg.Header.unmarshal(d.reader); err != nil {
+	if err := msg.Header.unmarshal(d.reader); err != nil {
 		return nil, err
 	}
 	// IPFIX Message Header validation
-	if err = msg.Header.validate(); err != nil {
+	if err := msg.Header.validate(); err != nil {
 		return nil, err
 	}
 
 	// Add source IP address as Agent ID
 	msg.AgentID = d.raddr.String()
 
+	// In case there are multiple non-fatal errors, collect them and report all of them.
+	// The rest of the received sets will still be interpreted, until a fatal error is encountered.
+	// A non-fatal error is for example an illegal data record or unknown template id.
+	var decodeErrors []error
 	for d.reader.Len() > 4 {
+		if err := d.decodeSet(mem, msg); err != nil {
+			switch err.(type) {
+			case nonfatalError:
+				decodeErrors = append(decodeErrors, err)
+			default:
+				return nil, err
+			}
+		}
+	}
 
-		setHeader := new(SetHeader)
-		setHeader.unmarshal(d.reader)
+	return msg, combineErrors(decodeErrors...)
```

Comment:
https://github.com/VerizonDigital/vflow/blob/master/vflow/ipfix.go#L225-L228
we need to change the logic at below to have the decoded set(s) once we have noneFatalError. 
```
		if decodedMsg, err = d.Decode(mCache); err != nil {
			logger.Println(err)
                        continue
		}
```
just by add a condition like below:
```
		if decodedMsg, err = d.Decode(mCache); err != nil {
			logger.Println(err)
                        if decodedMsg == nil { 
			     continue
                        }
		}
```
What do you think?